drivers/pwm/pwm_nxp_flexio.c

```diff
@@ -130,18 +132,35 @@ static int pwm_nxp_flexio_set_cycles(const struct device *dev,
 
 	pwm_info = &config->pulse_info->pwm_info[channel];
 
-	if ((flags & PWM_POLARITY_INVERTED) == 0) {
+	polarity = (flags & PWM_POLARITY_INVERTED) == 0 ?
+		FLEXIO_PWM_ACTIVE_HIGH : FLEXIO_PWM_ACTIVE_LOW;
+
+	/*
+	 * Checking to see if duty cycle is 0% or 100%
+	 * If so manually keep the GPIO HIGH or LOW.
+	 */
+	if (period_cycles == pulse_cycles) {
+		polarity = FLEXIO_PWM_ACTIVE_LOW;
+		is_only_low_or_high = true;
+	} else if (duty_cycle == period_cycles && pulse_cycles == 0) {
```

Comment:
now that I am looking at this, this looks like a lot of redundancy, all you are doing with the duty_cycles variable in reality based on the calculation is checking if pulse cycles is 0, you can just do some algebra to rearrange the equation to see that you are actually doing `pulse_cycles == 0 && pulse_cycles == 0` here. I don't think you need the duty_cycles variable at all.

Reply:
I will look at this more closely today.